build-tools/xaprepare/xaprepare/Steps/Step_Get_Android_BuildTools.cs

```diff
@@ -21,6 +21,16 @@ public Step_Get_Android_BuildTools ()
 			packages.Add ((package: $"build-tools_r{XABuildToolsVersion}-macosx.zip", prefix: XABuildToolsPackagePrefixMacOS));
 			packages.Add ((package: $"build-tools_r{XABuildToolsVersion}-windows.zip", prefix: XABuildToolsPackagePrefixWindows));
 			packages.Add ((package: $"build-tools_r{XABuildToolsVersion}-linux.zip", prefix: XABuildToolsPackagePrefixLinux));
+
+			// build-tools 30, for DX tests
+			string XABuildTools30Version     = Context.Instance.Properties [KnownProperties.XABuildTools30Version] ?? String.Empty;
+			string XABuildTools30PackagePrefixMacOS = Context.Instance.Properties [KnownProperties.XABuildTools30PackagePrefixMacOS] ?? string.Empty;
+			string XABuildTools30PackagePrefixWindows = Context.Instance.Properties [KnownProperties.XABuildTools30PackagePrefixWindows] ?? string.Empty;
+			string XABuildTools30PackagePrefixLinux = Context.Instance.Properties [KnownProperties.XABuildTools30PackagePrefixLinux] ?? string.Empty;
+
+			packages.Add ((package: $"build-tools_r{XABuildTools30Version}-macosx.zip", prefix: XABuildTools30PackagePrefixMacOS));
```

Comment:
Shouldn't this block be skipped if `XABuildTools30Version` or any prefix it uses are `String.Empty`? If the properties are optional, then there's no need to introduce potential errors in the build later on.

Reply:
I'll add a `string.IsNullOrEmpty()` check. I was thinking we'd just delete all this code when we stop supporting DX, but that might get missed or something.

Reply:
It'll be definitely safer this way